src/leaderboard/leaderboard.service.ts

```diff
@@ -2,12 +2,82 @@ import { AnalyticsDto } from '@/common/dto/leaderboard';
 import getLeaderBoardDataFROMJSON from '@/common/utils/getLeaderBoardDataFROMJSON';
 import { LeaderboardTypeAnalytics } from '@/types/leaderboard';
 import { Injectable, OnModuleInit } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import {
+  Leaderboard,
+  LeaderboardDocument,
+} from '@/common/mongoose/schemas/leaderboard';
 
 @Injectable()
 export class LeaderboardService implements OnModuleInit {
-  onModuleInit() {}
+  constructor(
+    @InjectModel(Leaderboard.name)
+    private readonly memberModel: Model<LeaderboardDocument>,
+  ) {}
+  async onModuleInit() {
+    await this.handleCron();
+  }
+
+  @Cron(CronExpression.EVERY_DAY_AT_MIDNIGHT)
+  async handleCron() {
+    console.log('Running fetch and store members cron job');
```

Comment:
That’s great @zushar! Very good implementation for the data insertion.

Before merging, I believe we need to make a few changes:

	1.	In the handleCron function, we should fetch new data only on Sundays. Fetching and processing this data daily is unnecessary. Please modify the CronExpression.EVERY_DAY_AT_MIDNIGHT to run every Sunday at 8:00 AM.
	2.	We need an additional function. Each time we fetch data, we retrieve all the records. While this may not be the most efficient method, it is the approach provided by GitHub. Consequently, this will accumulate a significant amount of data. Please add a function to delete all existing records before storing the new data as you currently do.

Let me know if you need any further adjustments!

Reply:
done